web/js/animation/gif.js

```diff
@@ -290,8 +291,8 @@ wv.anim.gif = wv.anim.gif || function (models, config, ui) {
    */
   var getOpacities = function (products) {
     var opacities = [];
-    _.each(products, function (product) {
-      opacities.push((_.isUndefined(product.opacity)) ? 1 : product.opacity);
+    loEach(products, function (product) {
+      opacities.push((loIsUndefined(product.opacity)) ? 1 : product.opacity);
```

Comment:
I believe that it is very helpful to know that the function is coming from Lodash. It informs you that if you want to see exactly how the function works, you can go look at the Lodash docs and that this method is not native to JS or some other library. There have been many occasions in the past where seeing the traditional `_.` has lead me to the lodash docs.

This is how we have been implementing it throughout the refactor and I think it is the best solution we have come up with.

Reply:
Does using the true name of the function prevent someone from looking up a function definition? Isn't a file with these contents clearly using `lodash`?
```
import each from 'lodash/each';
...
each(myArr, myFunc);
```
Why is the function name a useful place to communicate where the function is defined? Should native JavaScript functions all be prefixed with `native`, like `nativeConsole.log()`? Should we prefix our custom functions with the line number where we define them? `line66Build()`? I think any JavaScript developer knows how to look up a function definition if they need more information about how a function works. But when reading code, it's more useful for the function name to be intuitive and not include confusing conventions or noise.

What if we want to swap out a function with a different library in the future? What if we want to start using `underscore` or `Ramda` or define our own custom function? Then we'd have to rename the function just because the implementation changed, even if it's doing the same thing.

**In either case, a cryptic `lo` prefix doesn't clearly communicate to developers what's going on, and it's an anti-pattern that needs to be nipped in the bud.** Let's either use a more descriptive prefix and put the full name in, i.e. `lodashEach`, or create a `_` object that we attach functions to;

```
import each from 'lodash/each';
var _.each = each;
...
_.each(myArr, myFunc);

```

Reply:
I am not opposed to `_.each` or `lodashEach`. The thing is that we have already discussed this as a team multiple times and at least the majority of us agreed that it should be `loEach`. We are more than half way through a refactor and we have been using this approach the whole time. It doesn't seem like the time to worry about this issue.

Reply:
I agree this PR could be a tricky place to address this. We can open a separate PR that does a search and replace of lodash variable names in the entire code base at once, when we're closer to finishing the `module-loaders` PR.

I do think this refactor is the right time to worry about this because this refactor is establishing patterns that are going to be a part of our code base for years, and establishing a pattern of **clearly communicating** with variable names is extremely important.

We haven't reached a consensus as a team about which names to use (and arguing over the name of a variable is [bikeshedding](https://en.wiktionary.org/wiki/bikeshedding) anyway), but what I'm arguing for here is not the name of a single variable, it's that we use variable names that are clear and explicit, rather than ambiguous abbreviations. It's a matter of clear communication about the intent of our code. The code in Worldview is not meant to be read only by computers, it's also got to be read by all of us, by future us, by future developers who join the team, and by other open source contributors, and we need to make it as easy as possible for all of those people to understand the code base without unnecessary extra effort.

We should break the habit of using abbreviations in variable names, and instead prefer explicit variable names as a matter of clear communication.

Here's some good advice about naming variables: http://wiki.c2.com/?GoodVariableNames  My favorite quote from that page is;

> I tnk vrbl nms tht r abrvtd r hrd 2 rd, bt mby tht_s jst me.